internal/pkg/policy/self.go

```diff
@@ -201,31 +203,32 @@ func (m *selfMonitorT) updateStatus(ctx context.Context) (proto.StateObserved_St
 	m.mut.Lock()
 	defer m.mut.Unlock()
 
+	var err error
 	if m.policy == nil {
 		// no policy found
 		m.status = proto.StateObserved_STARTING
-		if m.policyId == "" {
-			m.reporter.Status(proto.StateObserved_STARTING, "Waiting on default policy with Fleet Server integration", nil)
+		if m.policyID == "" {
+			err = m.reporter.Status(proto.StateObserved_STARTING, "Waiting on default policy with Fleet Server integration", nil)
 		} else {
-			m.reporter.Status(proto.StateObserved_STARTING, fmt.Sprintf("Waiting on policy with Fleet Server integration: %s", m.policyId), nil)
+			err = m.reporter.Status(proto.StateObserved_STARTING, fmt.Sprintf("Waiting on policy with Fleet Server integration: %s", m.policyID), nil)
 		}
-		return proto.StateObserved_STARTING, nil
+		return proto.StateObserved_STARTING, err
```

Comment:
Should we be concerned about it returning a non-failed status with an error?

Reply:
it looks like the status update code doesn't return meaningful errors (i.e. always `nil`), but I like the idea for posterity.